src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1594,15 +1594,19 @@ private void CalculateIORedirection(bool isWindowsApplication, out bool redirect
                 //    $powershell.AddScript('ipconfig.exe')
                 //    $powershell.AddCommand('Out-Default')
                 //    $powershell.Invoke())
-                // we should not count it as a redirection.
-                if (IsDownstreamOutDefault(this.commandRuntime.OutputPipe))
+                // we should not count it as a redirection. Unless the native command has it's stdout redirected
+                // for example:
+                //    cmd.exe /c "echo test" > somefile.log
+                // in that case we want to keep output redirection even though Out-Default is the only
+                // downstream command.
+                if (IsDownstreamOutDefault(this.commandRuntime.OutputPipe) && StdOutDestination is null)
```

Comment:
I wonder if we didn't have tests covering the stdout redirection scenarios. Can you please add tests?